lib/Mojolicious/Plugin/DefaultHelpers.pm

```diff
@@ -81,7 +81,8 @@ sub _delay {
   my $c     = shift;
   my $tx    = $c->render_later->tx;
   my $delay = Mojo::IOLoop->delay(@_);
-  $delay->catch(sub { $c->helpers->reply->exception(pop) and undef $tx })->wait;
+  $delay->on(error => sub { $c->helpers->reply->exception(pop) and undef $tx });
```

Comment:
Why do you need to change this line? Doesn't it work the same way, since you use `pop` and not `$_[1]` ?

Reply:
At the time i didn't want to change behavior, promise callbacks are not the same as event callbacks. But it might be worth changing if `$c->delay` is also used purely for promises.